### DISCUSSION TITLE
🦔 v3.12.x

### DISCUSSION THREAD

**Discussion by bjlittle** [SciTools/iris#6134] [Releases]
<div align="center">

| Manager | Buddy | Milestone | Feature Branch | Project |
| :---: | :---: | :---: | :---: | :---: |
| @pp-mo | @ESadek-MO | [v3.12](https://github.com/SciTools/iris/milestone/52) | [v3.12.x](https://github.com/SciTools/iris/tree/v3.12.x) | 🦔 [v3.12.0](https://github.com/orgs/SciTools/projects/45/views/1) |

</div>

<div align="center">

| Minor Release | Status | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v3.12.0rc0` | [![status](https://img.shields.io/badge/v3.12.0rc0-released-green)](https://github.com/SciTools/iris/discussions/6134) | `10-03-2025` |[RTD](https://scitools-iris.readthedocs.io/en/v3.12.0rc0/) | [Release](https://github.com/SciTools/iris/releases/tag/v3.12.0rc0) | [conda-forge](https://anaconda.org/conda-forge/iris?version=3.12.0rc0)  | [PyPi](https://pypi.org/project/scitools-iris/3.12.0rc0/#files) |
| `v3.12.0` | [![status](https://img.shields.io/badge/v3.12.0-released-green)](https://github.com/SciTools/iris/discussions/6134) | `24-03-2025` | [RTD](https://scitools-iris.readthedocs.io/en/v3.12.0/) | [Release](https://github.com/SciTools/iris/releases/tag/v3.12.0) | [conda-forge](https://anaconda.org/conda-forge/iris?version=3.12.0) | [PyPI](https://pypi.org/project/scitools-iris/3.12.0/) |
| `v3.12.1` | [![status](https://img.shields.io/badge/v3.12.1-released-green)](https://github.com/SciTools/iris/discussions/6134) | `30-04-2025` | [RTD](https://scitools-iris.readthedocs.io/en/v3.12.1/) | [Release](https://github.com/SciTools/iris/releases/tag/v3.12.1) | [conda-forge](https://anaconda.org/conda-forge/iris?version=3.12.1) | [PyPI](https://pypi.org/project/scitools-iris/3.12.1/) |
| `v3.12.2` | [![status](https://img.shields.io/badge/v3.12.2-released-green)](https://github.com/SciTools/iris/discussions/6134) | `09-05-2025` | [RTD](https://scitools-iris.readthedocs.io/en/v3.12.2/) | [Release](https://github.com/SciTools/iris/releases/tag/v3.12.2) | [conda-forge](https://anaconda.org/conda-forge/iris?version=3.12.2) | [PyPI](https://pypi.org/project/scitools-iris/3.12.2/) |
| `v3.12.3` | [![status](https://img.shields.io/badge/v3.12.3-released-green)](https://github.com/SciTools/iris/discussions/6134) | `22-08-2025` | [RTD](https://scitools-iris.readthedocs.io/en/v3.12.3/) | [Release](https://github.com/SciTools/iris/releases/tag/v3.12.3) | [conda-forge](https://anaconda.org/conda-forge/iris?version=3.12.3) | [PyPI](https://pypi.org/project/scitools-iris/3.12.3/) |

</div>

**Comment by trexfeathers** [comment]
From discussion with @bjlittle and @pp-mo: we are wondering if we can focus this release on solving:

- #5165

**Comment by pp-mo** [comment]
@schlunma @bouweandela @valeriupredoi  just a heads-up ....
We have had some problems with progress in [3.12 sprint work](https://github.com/orgs/SciTools/projects/45) and are considering re-planning this release
  * delaying release dates by ~1 week.
  * **_not_** including a solution to https://github.com/SciTools/iris/issues/3678 which had been planned (https://github.com/SciTools/iris/pull/6190) -- since we have hit problems, and it is still unresolved

Otherwise, please 
  * refer to [3.12 sprint board](https://github.com/orgs/SciTools/projects/45/views/1) for what is planned
  * expect new dates of approx ... 
     * RC = 2025-03-10
     * 3.12.0 = 2025-03-24
     * (TBC shortly)
  * **Please respond if this is causing concern**

**Reply by valeriupredoi** [reply]
many thanks @pp-mo :beer: No concern at my end of (very DevOps-y technical) work, let me know if I can help in any way

**Reply by bouweandela** [reply]
I would prefer the delay and including a fix for https://github.com/SciTools/iris/issues/3678.

Any chance that a fix for #6188 can still be included? This is a pretty severe bug that affects cube and coordinate equality checks. I had a go at fixing it in #6325, but if you prefer to fix it in a different way that would also be great.

**Reply by pp-mo** [reply]
> I would prefer ... including a fix for https://github.com/SciTools/iris/issues/3678.

hmmm... well so would we, and it was prioritised, but at present it is just looking like a bit of a blocker.
I'll keep you posted, I think that's the best we can promise right now.

> Any chance that a fix for https://github.com/SciTools/iris/issues/6188 can still be included?

Well, I am looking into it again, but I need a bit of time ... so far trying to work out performance implications, whether they are serious and/or inevitable.

**Reply by bouweandela** [reply]
I looked at those a bit too and fixed the benchmarks in #6339 to get a more realistic report. I noticed a small inefficiency when looking at the profiling results that I think can be solved easily to compensate at least in part for the performance loss in the specific case used for the benchmark, so implemented that change in #6337.

**Reply by pp-mo** [reply]
## Update 2025-03-06
We are still planning to cut a 3.12 release on Monday, i.e. 2025-03-10.

Remaining tasks:
  * #6325 After some considerable more effort from @bouweandela , this is now merged
  * #3678 unfortunately is still very much stuck without an acceptable solution.  Though we do have a clear plan, it will just take more work
  * #6338 ..
  * .. _and_ #6334 : 
    * I hope to have these also completed by Monday, but they are also still not certain

**Comment by pp-mo** [comment]
Good new, the RC is now out !
Available on PyPI and conda-forge as per usual.
https://github.com/SciTools/iris/releases/tag/v3.12.0rc0

**Comment by pp-mo** [comment]
Now complete !
https://github.com/SciTools/iris/releases/tag/v3.12.0

**Comment by trexfeathers** [comment]
We have now released [Iris v3.12.1](https://github.com/SciTools/iris/releases/tag/v3.12.1).

This contains a single change, to make us compatible with Setuptools versions 80 and higher (a breaking incompatibility was discovered earlier this week #6424). This fix is needed for anyone that uses `pip install -e` with Iris, or any `setup.py` commands.

Ping @thebaptiste

**Reply by thebaptiste** [reply]
Many thanks !

**Comment by trexfeathers** [comment]
And now, we have released [Iris v3.12.2](https://github.com/SciTools/iris/releases/tag/v3.12.2)!

This makes Iris compatible with Dask `v2025.4.0` and higher, so you should no longer get any of those pesky errors about `Delayed` objects during loading and saving.

**Comment by trexfeathers** [comment]
Following an urgent user request: we have patched the 3.12 series to fix compatibility with Numpy >= v1.25 when comparing un-broadcastable arrays. This fix will be propagated to 3.13 and `main` in due course.

https://github.com/SciTools/iris/releases/tag/v3.12.3
